arrow/src/test/java/org/apache/calcite/adapter/arrow/ArrowData.java

```diff
@@ -89,6 +96,26 @@ private Schema makeArrowSchema() {
     return new Schema(childrenBuilder.build(), null);
   }
 
+  private Schema makeArrowTypeMetaDataSchema() {
+    ImmutableList.Builder<Field> childrenBuilder = ImmutableList.builder();
+    FieldType tinyIntType = FieldType.nullable(new ArrowType.Int(8, true));
+    FieldType smallIntType = FieldType.nullable(new ArrowType.Int(16, true));
+    FieldType intType = FieldType.nullable(new ArrowType.Int(32, true));
+    FieldType stringType = FieldType.nullable(new ArrowType.Utf8());
+    FieldType floatType =
```

Comment:
does arrow support both double and single precision for doubles?
Then I would do both at the same time.